tests/cypress/e2e/features/single_object_annotation.js

```diff
@@ -0,0 +1,161 @@
+    const frameCount = serverFiles.length;
+
+    let taskID = null;
+    let jobID = null;
+
+    const polygonPoints = [
+        { x: 300, y: 100 },
+        { x: 400, y: 400 },
+        { x: 400, y: 250 },
+        { x: 450, y: 350 },
+    ];
+
+    function createPolygon(points) {
+        points.forEach((element) => {
+            cy.get('.cvat-canvas-container').click(element.x, element.y);
+        });
+    }
+
+    function checkFrameNum(frameNum) {
+        cy.get('.cvat-player-frame-selector').within(() => {
+            cy.get('input[role="spinbutton"]').should('have.value', frameNum);
+        });
+    }
+
+    function submitJob() {
+        cy.get('.cvat-single-shape-annotation-submit-job-modal').should('exist');
+        cy.get('.cvat-single-shape-annotation-submit-job-modal').within(() => { cy.contains('Submit').click(); });
+
+        cy.intercept('PATCH', '/api/jobs/**').as('submitJob');
+        cy.wait('@submitJob').its('response.statusCode').should('equal', 200);
+
+        cy.get('.cvat-single-shape-annotation-submit-success-modal').should('exist');
+        cy.get('.cvat-single-shape-annotation-submit-success-modal').within(() => { cy.contains('OK').click(); });
+    }
+
+    function checkSingleShapeModeOpened() {
+        cy.get('.cvat-workspace-selector').should('have.text', 'Single shape');
+        cy.get('.cvat-canvas-controls-sidebar').should('not.exist');
+        cy.get('.cvat-player-frame-selector input').should('be.disabled');
+
+        cy.get('.cvat-single-shape-annotation-sidebar-hint').should('exist');
+        cy.get('.cvat-single-shape-annotation-sidebar-ux-hints').should('exist');
+    }
+
+    before(() => {
+        cy.visit('auth/login');
+        cy.login();
+        cy.headlessCreateTask({
+            labels: [{ name: labelName, attributes: [], type: 'polygon' }],
+            name: taskName,
+            project_id: null,
+            source_storage: { location: 'local' },
+            target_storage: { location: 'local' },
+        }, {
+            server_files: serverFiles,
+            image_quality: 70,
+            use_zip_chunks: true,
+            use_cache: true,
+            sorting_method: 'lexicographical',
+        }).then((response) => {
+            taskID = response.taskID;
+            [jobID] = response.jobIDs;
+        }).then(() => {
+            cy.visit(`/tasks/${taskID}/jobs/${jobID}`);
+            cy.get('.cvat-canvas-container').should('exist').and('be.visible');
+        });
+    });
+
+    after(() => {
+        cy.logout();
+        cy.getAuthKey().then((response) => {
+            const authKey = response.body.key;
+            cy.request({
+                method: 'DELETE',
+                url: `/api/tasks/${taskID}`,
+                headers: {
+                    Authorization: `Token ${authKey}`,
+                },
+            });
+        });
+    });
+
+    describe('Tests basic features of single shape annotation mode', () => {
```

Comment:
I will propose to add mode `describe(` for other shapes. It will be almost copy paste.

Reply:
Not sure we need to check all the controls for each shape, but we should check that these shapes may be drawn